bp_fe/src/include/bp_fe_top.vh

```diff
@@ -0,0 +1,70 @@
+/*
+ * bp_fe_top.vh provides all the necessary structs for the Frontend submodules.
+ * Backend supplies the frontend with branch prediction results and exceptions
+ * codes. The Frontend should update the states accordingly.
+*/
+
+`ifndef BP_FE_TOP_VH
+`define BP_FE_TOP_VH
+
+`ifndef BP_COMMON_FE_BE_IF
+`define BP_COMMON_FE_BE_IF
+`include "bp_common_fe_be_if.vh"
+`endif
+
+/*
+ * bp_fe_pc_gen_s provides the interfaces of the Frontend to pc_gen. The interfaces
+ * consist of all the necessary information from Backend to the Frontend, including 
+ * subopcodes, branch metadata , misprediction reasons, and whether translation is
+ * needed for virtual to physical addresses.
+*/
+
```

Comment:
If there are no parameters for the struct, just declare it without a macro.  If it's dependent on something that is parameterized, declare it in the same macro see https://github.com/black-parrot/pre-alpha-release/blob/be_dev/bp_common/bp_common_fe_be_if.vh